src/components/form/utils/use_intent.ts

```diff
@@ -0,0 +1,5 @@
+import type { Intent } from '@blueprintjs/core';
+
+export function useIntent(error?: string): Intent | undefined {
```

Comment:
It's misleading to name this like a hook because it is not one.

Reply:
i renamed it to "getIntent"